x-pack/plugins/security_solution/public/detections/components/rules/description_step/index.tsx

```diff
@@ -204,12 +205,29 @@ export const getDescriptionItem = (
   } else if (field === 'responseActions') {
     return [];
   } else if (field === 'groupByFields') {
+    // only query rule can have suppression group by fields
+    const ruleType: Type = get('ruleType', data);
+    if (!isQueryRule(ruleType)) {
+      return [];
+    }
     const values: string[] = get(field, data);
     return buildAlertSuppressionDescription(label, values);
   } else if (field === 'groupByRadioSelection') {
     return [];
   } else if (field === 'groupByDuration') {
-    if (get('groupByFields', data).length > 0) {
+    const ruleType: Type = get('ruleType', data);
+    // only query and threshold rules can have duration property
+    if (!isQueryRule(ruleType) && !isThresholdRule(ruleType)) {
```

Comment:
Nit: you could convert this comment into code (which would make it less likely for the comment to become outdated in the future):
```suggestion
    const ruleCanHaveDuration = isQueryRule(ruleType) || isThresholdRule(ruleType);
    if (!ruleCanHaveDuration) {
```